OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -130,16 +141,24 @@ public override void Tick(World world)
 
 		public override string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
 		{
-			var prefix = mi.Modifiers.HasModifier(Modifiers.Ctrl) ? "assaultmove" : "attackmove";
-
 			if (world.Map.Contains(cell) && subjects.Any())
 			{
-				var explored = subjects.First().Actor.Owner.Shroud.IsExplored(cell);
+				var subject = subjects.First();
+				var explored = subject.Actor.Owner.Shroud.IsExplored(cell);
 				var blocked = !explored && subjects.Any(a => !a.Trait.Info.MoveIntoShroud);
-				return blocked ? prefix + "-blocked" : prefix;
+				var info = subject.Trait.Info;
+				var prefix = mi.Modifiers.HasModifier(Modifiers.Ctrl);
+				if (prefix && !blocked)
+					return info.AssaultMoveCursor;
+				if (prefix && blocked)
+					return info.AssaultMoveBlockedCursor;
+				if (prefix && !blocked)
+					return info.AttackMoveCursor;
+				if (prefix && blocked)
+					return info.AttackMoveBlockedCursor;
 			}
 
-			return prefix + "-blocked";
+			return "default";
```

Comment:
This doesn't look correct.

Reply:
I don't have a solution for it and it doesn't seem to regress in-game.

Reply:
The (attack)move-blocked cursor will no longer show when hovering outside of the map bounds.

Reply:
I reworked this.

Reply:
This still doesn't look correct. We shouldn't be introducing "default" at all.

Reply:
I can't access the `subject.Trait.Info` from below here so this fallback though probably never hitting is required.

Reply:
Return `null` so that the engine has a chance to substitute a cursor from a different unit in the selection.

Reply:
Okay.